### ISSUE TITLE
ngClass and [class] works bad together

### ISSUE THREAD

**Issue by sur-run** [angular/angular#15396]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```
I'm trying to use both ngClass and [class] to set classes that a need like here
`<h1 [ngClass]="{'ui-true-class': true, 'ui-false-class': false}" [class]="styleClass">Hello</h1>
    <button (click)="styleClass='styleClassValue'">set styleClass</button>`

**Current behavior**

When some value setted to styleClass variable it also writes all key-classes from ngClass'es map that has false value(ui-false-class in example)
**Expected behavior**

Setting value to styleClass doesn't affect classes from ngClass value like described above
**Minimal reproduction of the problem with instructions**
you can see it here [plunk](http://plnkr.co/edit/HPpLytOFQr19UoBp9lrD?p=preview)
1. see initial value class on element "Hello" is "ui-true-class"
2. click "set styleClass" button
3. see value class on element "Hello" is "ui-true-class styleClassValue ui-false-class"

I've tried to look deeper and found that after changing styleClass value like in example classes from ngClass setted after this sequence of method calls:
set ng_class.js
NgClass._applyClasses(this._rawClass, _false_) ng_class.js
and code here
Object.keys(rawClassVal).forEach(function (klass) {
                    if (**isPresent(rawClassVal[klass])**)
                        _this._toggleClass(klass, !_isCleanup_);
                });

So rawClassVal[klass] by itself doesnt matter here and klass always will toggle since second parameter in 
NgClass._applyClasses(which called isCleanup) equals 'false'.

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

* **Angular version:** 2.0.X

* **Browser:** [Chrome | Firefox ] at least

 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

* **Node (for AoT issues):** `node --version` =

**Comment by pkozlowski-opensource** [comment]
Using `[class]` and `[ngClass]` together is asking for trouble as you've got multiple bindings to the same property. I will have a look at this one to see if we can have sth reasonable here but I would suggest moving class calculation logic to a function and using `ngClass` exclusively

**Comment by vytautas-pranskunas-** [comment]
I noticed same behavior with interpolation like:
<div class="some-class-{{someVariable}}" [class.new-class]="true">

new-class is not added

**Comment by vicb** [comment]
As @pkozlowski-opensource said, you should not use them at the same time.

**Comment by artuska** [comment]
+1

Have the same problem with class interpolation + ngClass:

```
<button class="button button--medium button--special button--special--{{type}}" [ngClass]="{'processing': isProcessing}">
    <span class="button__text button__text--medium">
        <span>Подключить</span>
    </span>
</button>
```

When the `type` changes button gets `processing` class, wtf? `isProcessing` is always set to `false`!

**Comment by pkozlowski-opensource** [comment]
@artuska it should have been fixed from Angular 6.0.0-rc.2 via 4a426696c98b28d20175cc3244ffa8b8b703790f. 

Please open a new issue with a minimal reproduce scenario (plunker / stackblitz) if you still face problems with Angular 6.
